### ISSUE TITLE
Performance degradation after upgrade 1.93.6 => 1.97.1

### ISSUE THREAD

**Issue by evkuzin** [VictoriaMetrics/VictoriaMetrics#6211]
### Describe the bug

After upgrade one of the nodes I see unhealthy spikes on the graphs. The fleet is running 1.93.6 and only one storage node was upgraded to 1.97.1.
After upgrade from time to time I will see a significant spike in pending datapoints for storage (baseline is 700k/1.5 mil, spikes on the new version up to 20 mil) 

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/16777863/dd9b64b8-baa8-49d5-a124-72cb9c2af689)

The ingection rates from storage metric on the other side stays the same for the whole fleet 
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/16777863/ab6a952c-0e56-4b49-be62-f571c4d0c069)

Also there is a big difference in inmemory merges speed for new vs old version. See graph (which correlates with the spikes for pending datapoints BTW
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/16777863/48b34a1a-7d24-4ff7-b420-5ad53fa4f1d4)

### To Reproduce

I'm not sure how to repro honestly, I suppose just install two these versions? 

### Version

```
ssh rd1riamet2 /usr/bin/vmstorage-prod -version
vmstorage-1.93.6
ssh rd1riamet1 /usr/bin/vmstorage-prod -version
vmstorage-1.97.1
```

### Used command-line flags

```
ExecStart=/usr/bin/vmstorage-prod -httpListenAddr ':8482' -storageDataPath /var/lib/victoriametrics -retentionPeriod 1825d -vminsertAddr ':8400' -vmselectAddr ':8401' -dedup.minScrapeInterval 10s
```

**Comment by hagen1778** [comment]
Hello @evkuzin! 
It is likely related to more agressive background merge policy change in 1.97:
> FEATURE: make [background merge](https://docs.victoriametrics.com/#storage) more responsive and scalable. This should help the following issues: [5190](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5190), [3425](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3425), [648](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/648).

Can you check whether number of background merges or their speed increased for `big` parts?

**Comment by evkuzin** [comment]
Oddly enough its 0 for last 24 hours on the new instance comparing to old version
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/16777863/b98cc91b-7ae2-4114-a81a-8ee4f41a6aa4)

**Comment by hagen1778** [comment]
I can't correlate this graph with previous ones as they have different X axis. Could you correlate and see of performance degradation is connected to increased merge activities?

> Oddly enough its 0 for last 24 hours on the new instance comparing to old version

Apparently, the mentioned change unblocked some of the merges which were postponed for historical data. Newer instances have less historical data, so they have less data to merge or no data at all.

**Comment by evkuzin** [comment]
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/16777863/f3904b29-2f88-4eee-aefa-41978abf3019)
Same timeline for better visibility. 
But yeah, regarding 
> Could you correlate and see of performance degradation is connected to increased merge activities?
See first post 1 and third picture. 
It does correlate - when pending datapoints are bursting to 20 millions merge speed is 3 mil on the same node and when merge speed is down to 2 mil, pending datapoint's are down to a base threshold (by base I mean similar to old version instances)

**Comment by evkuzin** [comment]
In the telegram chat it was mentioned that its okay to have ingestion rate <= x2 pending datapoints. But here we have x10 and this is bothering me.

**Comment by hagen1778** [comment]
> In the telegram chat it was mentioned that its okay to have ingestion rate <= x2 pending datapoints. But here we have x10 and this is bothering me.

I think the metric of pending datapoints became misleading with time. It doesn't illustrate if system is overloaded or not. Although, changes to background merges logic obsoleted the comment about x2 of ingestion rate. If you check the [playground](https://play-grafana.victoriametrics.com/d/oS7Bi_0Wz_vm/victoriametrics-cluster-vm?orgId=1&viewPanel=14) it shows pending rows 1Mil and ingestion rate 200k/s, so x5. 

Let's sum up the issue:
1. It is expected that after update from older versions to 1.97+ VM will run additional merges. These merges should finish shortly and won't bother you again.
2. Pending samples metric is misleading and will be likely removed from the Grafana dashboard. If there are no other signs of performance degradation, you can ignore pending samples metric.
